src/components/settings/Account/Billing/ManageButton.tsx

```diff
@@ -24,59 +24,70 @@ export const ManageBillingButton: React.FC<ManageBillingButtonProps> = ({
   buttonProps = { variant: "outline", color: loading ? "gray" : "blue.500" },
   onClick,
 }) => {
-  const [sessionUrl, setSessionUrl] = useState();
+  const [sessionUrl, setSessionUrl] = useState(account.stripePaymentActionUrl);
   const mutation = useCreateBillingSession();
-  const validPayment = account.status === AccountStatus.ValidPayment;
-  const paymentVerification =
-    account.status === AccountStatus.PaymentVerification;
-  const invalidPayment = [
-    AccountStatus.InvalidPayment,
-    AccountStatus.InvalidPaymentMethod,
-  ].includes(account.status);
+
+  let [buttonLabel, buttonText]: string[] = ["", ""];
+  switch (account.status) {
+    case AccountStatus.InvalidPayment:
+    case AccountStatus.ValidPayment: {
+      buttonLabel = "manage";
+      buttonText = "Manage billing";
+      break;
+    }
+    case AccountStatus.PaymentVerification: {
+      buttonLabel = "verifyPaymentMethod";
+      buttonText = "Verify your payment method →";
+      break;
+    }
+    case AccountStatus.InvalidPaymentMethod: {
+      buttonLabel = "addAnotherPayment";
+      buttonText = "Add another payment method →";
+      break;
+    }
+    default: {
+      buttonLabel = "addPayment";
+      buttonText = "Add payment method";
+      break;
+    }
+  }
```

Comment:
this... technically works but it would be a lot cleaner if you just wrapped it in either a `useMemo()` or just an iife, you wouldn't have to do the `let` + assign after dance

Reply:
That makes sense to me, never thought of it that way. I updated it to use a `useMemo()`, thanks Jonas! ✅